student-store-ui/src/components/App/App.jsx

```diff
@@ -1,21 +1,108 @@
-import * as React from "react"
-import { BrowserRouter } from 'react-router-dom'
-import Navbar from "../Navbar/Navbar"
-import Sidebar from "../Sidebar/Sidebar"
-import Home from "../Home/Home"
-import "./App.css"
+import * as React from "react";
+import { BrowserRouter, Link, Route, Routes } from "react-router-dom";
+import Navbar from "../Navbar/Navbar";
+import Sidebar from "../Sidebar/Sidebar";
+import Home from "../Home/Home";
+import History from "../History/History";
+import { useEffect, useState } from "react";
+import axios from "axios";
+import "./App.css";
+import ProductDetail from "../ProductDetail/ProductDetail";
+import NotFound from "../NotFound/NotFound";
 
 export default function App() {
+  const url = "http://localhost:3001/store";
+  const [products, setProducts] = useState([]);
+  const [shoppingCart, setShoppingCart] = useState([]);
+
+  // fetch the data
+  useEffect(() => {
+    axios
+      .get(url)
+      .then((response) => {
+        setProducts(response.data.products);
+      })
+      .catch((error) => {
+        setProducts([]);
+      });
+  }, []);
+
+  const handleAddItemToCart = (id) => {
+    let index = -1;
+
+    // find if the element exists
+    shoppingCart.forEach((element, i) => {
+      element.itemId === id ? (index = i) : undefined;
+    });
+
+    if (index < 0) {
+      shoppingCart.push({ itemId: id, quantity: 0 });
```

Comment:
This mutates state directly. It may not have been an issue with this project but it is an anti pattern in React and can lead to bugs. Instead, you should write this line as below:

```
setShoppingCart([...shoppingCart, {itemId: id, quantity: 0}])